bin/compile

```diff
@@ -1,62 +1,48 @@
+  echo '       !!!! See https://help.github.com/articles/creating-an-oauth-token-for-command-line-use' >&2
   exit 1
-end
+fi
 
-netrc_template = File.read "#{File.join(File.dirname(__FILE__), '..', 'support', 'netrc')}"
+GITHUB_AUTH_TOKEN="$(cat "${ENV_DIR}/GITHUB_AUTH_TOKEN")"
 
-##
-# Supply `curl` on the BUILD machine (not in slug) with token to access private Github repos
-File.open "#{ENV['HOME']}/.netrc", "w+", 0600 do |f|
-  # The ../support/netrc template relies on the `github_token` variable above
-  f.puts ERB.new(netrc_template).result
-end
+# Set the credentials in `.netrc` for build time only (they are intentionally not put in the slug).
+echo "machine github.com login ${GITHUB_AUTH_TOKEN} password x-oauth-basic" >> "${HOME}/.netrc"
+echo "machine api.github.com login ${GITHUB_AUTH_TOKEN} password x-oauth-basic" >> "${HOME}/.netrc"
 
-##
-# Tell `curl` on the BUILD machine (not in slug) to use ~/.netrc
-File.open "#{ENV['HOME']}/.curlrc", "w+", 0600 do |f|
-  f.puts "--netrc-optional"
-end
-
-if valid_login? github_token
-  puts "       Generated .netrc & .curlrc files (available only at build-time)"
-
-  ##
-  # Remove $GITHUB_AUTH_TOKEN from runtime environment at slug boot time.
-  FileUtils.mkdir_p "#{build_dir}/.profile.d"
-  File.open "#{build_dir}/.profile.d/netrc.sh", "w+", 0600 do |f|
-    f.puts "unset GITHUB_AUTH_TOKEN"
-  end
-
-  ##
-  # Print out user information
-  puts user_block(github_token)
-else
-  puts "       !!!! No valid Github user found with GITHUB_AUTH_TOKEN: '#{github_token}'"
+# Configure `curl` to use `.netrc` (again at build time only).
+echo '--netrc-optional' >> "${HOME}/.curlrc"
+
+# Unset the `$GITHUB_AUTH_TOKEN` env var at runtime, so it's not exposed to the app.
+mkdir -p "${BUILD_DIR}/.profile.d"
+echo 'unset GITHUB_AUTH_TOKEN' > "${BUILD_DIR}/.profile.d/netrc.sh"
+
+if ! USERNAME="$(curl ${CURL_OPTIONS} https://api.github.com/user | jq --raw-output .login)"; then
+  echo '       !!!! Failed to retrieve user information from GitHub. Is `GITHUB_AUTH_TOKEN` set to a valid token?' >&2
+  echo '       !!!! See https://help.github.com/articles/creating-an-oauth-token-for-command-line-use' >&2
   exit 1
-end
+fi
+
+if ! ORGS="$({ curl ${CURL_OPTIONS} https://api.github.com/user/orgs | jq --raw-output 'map(.login) | sort | join(", ")'; } 2> /dev/null)"; then
+  ORGS="Unknown (Token needs at least 'read:org' or 'user' scopes to list organizations)"
+fi
+
+echo '       Generated .netrc & .curlrc files (available only at build-time)'
+echo "       GitHub User:   ${USERNAME}"
+echo "       Organizations: ${ORGS}"
 
 ##
-# Set github config to always use https
-`git config --global url."https://".insteadOf git://`
\ No newline at end of file
+# Set GitHub config to always use https
+`git config --global url."https://".insteadOf git://`
+# This additional GitHub config will help for using HTTPS instead of SSH for npm versions 7+
+`git config --global url."https://github.com".insteadOf ssh://git@github.com`
```

Comment:
This is the line I added.  I found this issue in Travis (I'd already set it locally), and this extra line will help "ssh" type urls from a package-lock.json or package.js be coerced into "https" type urls.